.github/workflows/main.yml

```
@@ -41,6 +41,9 @@ jobs:
         env:
           RUSTDOCFLAGS: -D warnings
 
+      - name: Install targets
+        run: rustup target add i686-unknown-linux-gnu wasm32-unknown-unknown
+
```

Even though the step below has argument `--all-targets`, that means a [different kind of target](https://doc.rust-lang.org/cargo/commands/cargo-check.html#target-selection)[^clippydoc], like `lib` or `bin` or `example`. To actually use the rust toolchains for the 32-bit architectures, this step could instead be:

```suggestion
      - name: Build for 32-bit architectures
        run: |
          rustup target add i686-unknown-linux-gnu wasm32-unknown-unknown
          cargo build --target i686-unknown-linux-gnu
          cargo build --target wasm32-unknown-unknown
```

I'm not sure it's a good idea to include this step is the `main.yml` workflow, as it is run on all three major operating systems, creating duplicate work and slowing down CI. It might be better to add a new CI workflow instead. Alternatively, this step can execute conditional to the current `matrix.os`.

[^clippydoc]: `cargo clippy --help` points to `cargo check --help` which is equivalent to the linked page.

---

Thanks for the correction. How do you recommend to proceed?

---

The first thing that comes to mind is to create a new CI workflow specifically for building the crate on a 32-bit architecture. This keeps the individual workflows _relatively_ lean, performs the new step only once instead of once per element in the strategy's matrix, at the expense of having a new workflow. Such a workflow could look like this:

```yaml
name: 32bit

on:
  push:
    branches:
      - master
  pull_request:
    branches:
      - master

jobs:
  build:
    name: Build for 32-bit targets
    runs-on: ubuntu-latest
    steps:
      - name: Checkout repository
        uses: actions/checkout@v4

      - name: Install stable toolchain
        uses: dtolnay/rust-toolchain@stable

      - name: Build for 32-bit architectures
        run: |
          rustup target add i686-unknown-linux-gnu wasm32-unknown-unknown
          cargo build --target i686-unknown-linux-gnu
          cargo build --target wasm32-unknown-unknown

```